pkg/svcat/service-catalog/instance.go

```diff
@@ -34,21 +34,50 @@ const (
 )
 
 // RetrieveInstances lists all instances in a namespace.
-func (sdk *SDK) RetrieveInstances(ns string) (*v1beta1.ServiceInstanceList, error) {
+func (sdk *SDK) RetrieveInstances(ns, classFilter, planFilter string) (*v1beta1.ServiceInstanceList, error) {
 	instances, err := sdk.ServiceCatalog().ServiceInstances(ns).List(v1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("unable to list instances in %s (%s)", ns, err)
 	}
 
-	return instances, nil
+	if classFilter == "" && planFilter == "" {
+		return instances, nil
+	}
+
+	filtered := v1beta1.ServiceInstanceList{
+		Items: []v1beta1.ServiceInstance{},
+	}
+
+	for _, instance := range instances.Items {
+		if classFilter != "" && instance.Spec.GetSpecifiedClusterServiceClass() != classFilter {
+			continue
+		}
+
+		if planFilter != "" && instance.Spec.GetSpecifiedClusterServicePlan() != planFilter {
+			continue
+		}
+
+		filtered.Items = append(filtered.Items, instance)
+	}
+
+	return &filtered, nil
 }
 
 // RetrieveInstance gets an instance by its name.
-func (sdk *SDK) RetrieveInstance(ns, name string) (*v1beta1.ServiceInstance, error) {
+func (sdk *SDK) RetrieveInstance(ns, name, classFilter, planFilter string) (*v1beta1.ServiceInstance, error) {
```

Comment:
This function shouldn't have been modified to accept a class and plan filter. It retrieves a single instance by name, so filtering doens't really apply here.

How about we add check to the `Validate` method on the commands that checks for the name and the class/plan filters being set at the same time.